src/utilities.jl

```diff
@@ -145,3 +145,73 @@ end
 
 check_positive_definite(M::AbstractMatrix; kwargs...) =
   check_positive_definite(LinearOperator(M); kwargs...)
+
+
+  """
+  solve_shifted_system!(B, 
+                        x, 
+                        σ, 
+                        γ_inv, 
+                        inv_Cx, 
+                        p,
+                        v, 
+                        u)
+
+Solves linear system (B + σI) x = b, where B is a forward L-BFGS operator and σ ≥ 0.
+
+### Parameters
+
+- `B::LBFGSOperator{T,I,F1,F2,F3}`: forward L-BFGS operator.
+- `x::AbstractVector{T}`: The vector representing `-b`.
+- `σ::T`: Nonnegative shift.
+- `inv_Cx::AbstractVector{T}`: A preallocated vector used to store the solution.
+
+### Returns
+
+- `inv_Cx::AbstractVector{T}`: The solution vector `x`, the solution to `(B + σI) x = b`.
```

Comment:
This is very confusing. There's already an argument called x. Isn't that the solution?